src/Services/Signatures/HashSignature.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace LearnositySdk\Services\Signatures;
+
+use LearnositySdk\Exceptions\ValidationException;
+
+class HashSignature implements SignatureInterface
+{
+    const ALGORITHM = 'sha256';
+
+    const SIGNATURE_VERSION = '01';
+
+    const CONSUMER_KEY_LENGTH = 16;
+
+    const TIMESTAMP_KEY_LENGTH = 13;
+
+    const SIGNATURE_KEY_LENGTH = 64;
+
+    const EXCEPTION_MESSAGE =
+        'The pre hash string for this signature type must contain the secret key';
+
+    /**
+     * @param string $preHashString
+     * @param string $secretKey
+     * @return string
+     * @throws ValidationException
+     */
+    public static function sign(
+        string $preHashString,
+        string $secretKey
+    ): string {
+        if (!strpos($preHashString, $secretKey)) {
+            throw new ValidationException(self::EXCEPTION_MESSAGE);
+        }
+        return hash(self::ALGORITHM, $preHashString);
```

Comment:
Best to use `static` rather than `self` everywhere, as the former refers to the current class (even if it's a descendant from your class here), while self refers to the class where it's define, which will always be this one.

I had a quick example here: https://gist.github.com/shtrom/b98252b215346c625a449010c10a2b4f